modPrecompiled/test/org/aion/precompiled/contracts/DummyRepo.java

```diff
@@ -122,7 +122,7 @@ public ByteArrayWrapper getStorageValue(Address addr, ByteArrayWrapper key) {
         Map<String, byte[]> map = storage.get(addr);
         if (map != null && map.containsKey(key.toString())) {
             byte[] res = map.get(key.toString());
-            if (res.length == DataWord.BYTES) {
+            if (res.length <= DataWord.BYTES) {
```

Comment:
Where do we have instances that do not match the specified length?

Reply:
It was previous the repository's duty to take a `DataWord` and truncate it's leading zeroes, then insert it. Now the repository does nothing and this truncating burden is on the caller. Likewise, the caller now also inherits the burden of having to re-pad this data, and that's what's being done here. This catches any truncated data words and re-pads them